src/pages/DepositSwiftPage/index.tsx

```diff
@@ -0,0 +1,253 @@
+import * as classnames from 'classnames';
+import {Field, FieldProps, Form, Formik, FormikProps} from 'formik';
+import {inject, observer} from 'mobx-react';
+import * as React from 'react';
+import * as CopyToClipboard from 'react-copy-to-clipboard';
+import {RouteComponentProps} from 'react-router-dom';
+import Yup from 'yup';
+import {RootStoreProps} from '../../App';
+import {AmountInput} from '../../components/AmountInput';
+import ClientDialog from '../../components/ClientDialog';
+import WalletTabs from '../../components/WalletTabs/index';
+import {
+  ROUTE_DEPOSIT_SWIFT_EMAIL_SENT,
+  ROUTE_WALLETS_TRADING
+} from '../../constants/routes';
+import {STORE_ROOT} from '../../constants/stores';
+import {DepositSwiftModel, DialogModel} from '../../models';
+import {DialogConditionType} from '../../models/dialogModel';
+
+import './style.css';
+
+interface DepositSwiftPageProps
+  extends RootStoreProps,
+    RouteComponentProps<any> {}
+
+export class DepositSwiftPage extends React.Component<DepositSwiftPageProps> {
+  readonly assetStore = this.props.rootStore!.assetStore;
+  readonly depositStore = this.props.rootStore!.depositStore;
+  readonly dialogStore = this.props.rootStore!.dialogStore;
+
+  componentDidMount() {
+    const {assetId} = this.props.match.params;
+    this.depositStore.fetchSwiftRequisites(assetId);
+
+    const clientDialog = this.dialogStore.pendingDialogs.find(
+      (dialog: DialogModel) =>
+        dialog.conditionType === DialogConditionType.Predeposit
+    );
+    if (clientDialog) {
+      clientDialog.visible = true;
+    }
+
+    window.scrollTo(0, 0);
+  }
+
+  render() {
+    const {assetId} = this.props.match.params;
+    const asset = this.assetStore.getById(assetId);
+    const sendSwiftRequisites = this.depositStore.sendSwiftRequisites;
+    const onSubmitSuccess = () => {
+      this.props.history.replace(ROUTE_DEPOSIT_SWIFT_EMAIL_SENT);
+    };
+    const clientDialog = this.dialogStore.pendingDialogs.find(
```

Comment:
You have same code, it means that it make sense to create helper function is dialogStore

Reply:
I'm not sure that every filtered subset should be moved to separate function because there are a lot of filtering variations and it will be mess with a lot of helper functions with different filtering options

Reply:
e.g. for now we will have predepositDialogs then hiddenDialogs then hiddenPredepositDialogs it's really weird for me